lib/date_range.rb

```diff
@@ -0,0 +1,54 @@
+require_relative "reservation_booker"
+require 'date'
+
+module Hotel
+    class DateRange
+      def self.valid_checkin_dates?(check_in_year, check_in_month, check_in_day)
+        is_valid = true
+        if !Date.valid_date?(check_in_year.to_i, check_in_month.to_i, check_in_day.to_i) 
+            is_valid = false
+        end
+        return is_valid
+        end
+
+        def self.valid_checkout_dates?(check_out_year, check_out_month, check_out_day)
+            is_valid = true
+            if !Date.valid_date?(check_out_year.to_i, check_out_month.to_i, check_out_day.to_i)
+                is_valid = false
+            end
+            return is_valid
+        end
+
+        def self.check_in_date(check_in_year, check_in_month, check_in_day)
```

Comment:
Again, let the date class do this work for you. All you've done here is make a small wrapper.